test/QueryToSocketDistributorTest.cpp

```diff
@@ -74,31 +74,33 @@ ASYNC_TEST(QueryToSocketDistributor, doubleSignalEndThrowsError) {
 
 ASYNC_TEST(QueryToSocketDistributor, signalEndWakesUpListeners) {
   auto ptr = std::make_shared<QueryToSocketDistributor>(ioContext, noop);
-  QueryToSocketDistributor& queryToSocketDistributor = *ptr;
-  auto impl = [&]() -> net::awaitable<void> {
-    bool waiting = false;
-    auto listener = [&]() -> net::awaitable<void> {
-      waiting = true;
-      EXPECT_EQ(nullptr,
-                co_await queryToSocketDistributor.waitForNextDataPiece(0));
-    };
-
-    auto broadcaster = [&]() -> net::awaitable<void> {
-      // Ensure correct order of execution
-      ASSERT_TRUE(waiting);
-      queryToSocketDistributor.signalEnd();
-      co_return;
-    };
-    co_await (listener() && broadcaster());
-  };
-  co_await net::co_spawn(queryToSocketDistributor.strand(), impl,
-                         net::use_awaitable);
+  co_await net::co_spawn(
+      ptr->strand(),
+      [](auto ptr) -> net::awaitable<void> {
+        QueryToSocketDistributor& queryToSocketDistributor = *ptr;
+        bool waiting = false;
+        auto listener = [&]() -> net::awaitable<void> {
+          waiting = true;
+          EXPECT_EQ(nullptr,
+                    co_await queryToSocketDistributor.waitForNextDataPiece(0));
+        };
+
+        auto broadcaster = [&]() -> net::awaitable<void> {
+          // Ensure correct order of execution
+          ASSERT_TRUE(waiting);
+          queryToSocketDistributor.signalEnd();
+          co_return;
+        };
+        co_await (listener() && broadcaster());
+      }(ptr),
+      net::deferred);
```

Comment:
By "inlining" I meant to pass `listener() && broadcaster()` to `net::co_spawn` directly instead of awaiting it inside a coroutine